etl/storage/storage.go

```diff
@@ -103,6 +104,26 @@ func filterByPulse(query *gorm.DB, pulseNumberLt, pulseNumberGt *int) *gorm.DB {
 	return query
 }
 
+func filterByPulseNumber(query *gorm.DB, pulseNumberLte, pulseNumberLt, pulseNumberGte, pulseNumberGt *int) *gorm.DB {
```

Comment:
better idea was to calculate query argument on parsing parameters phase. To have 'filter' object and pass it here.